src/ui/connectionmanager.cc

```diff
@@ -0,0 +1,205 @@
+  server_process_->deleteLater();
+  server_process_ = nullptr;
+}
+
+void ConnectionManager::connectionDialogAccepted() {
+  if(connection_dialog_->runANewServer()) {
+    startLocalServer();
+  }
+
+  // TODO actually connect with the server
+  // Can be done once C++ client library is created.
+}
+
+void ConnectionManager::startLocalServer() {
+  if(server_process_) {
+    server_process_->deleteLater();
+  }
+
+  server_process_ = new QProcess(this);
+  server_process_->setProcessChannelMode(QProcess::MergedChannels);
+  connect(server_process_, &QProcess::started,
+      this, &ConnectionManager::locallyCreatedServerStarted);
+  connect(server_process_, static_cast<void(QProcess::*)
+      (int, QProcess::ExitStatus)>(&QProcess::finished),
+      this, &ConnectionManager::locallyCreatedServerFinished);
+  connect(server_process_, &QIODevice::readyRead,
+        this, &ConnectionManager::serverProcessReadyRead);
+
+  QString server_file_path = connection_dialog_->serverScript();
+  QFileInfo server_file(server_file_path);
+  QString directory_path = server_file.absoluteDir().path();
+  QString server_file_name = server_file.fileName();
+
+  server_process_->setWorkingDirectory(directory_path);
+  QProcessEnvironment env = QProcessEnvironment::systemEnvironment();
+  env.insert("PYTHONUNBUFFERED", "1");
+  server_process_->setProcessEnvironment(env);
+
+  QStringList arguments;
+  arguments
+      << server_file_name
+      << connection_dialog_->databaseFile()
+      << QString("%1:%2").arg(connection_dialog_->serverHost())
+      .arg(connection_dialog_->serverPort());
+
+#if defined(Q_OS_LINUX)
+  QString python_interpreter_executable("python3");
+#elif defined(Q_OS_WIN)
+  QString python_interpreter_executable("py.exe");
+#else
+  QString python_interpreter_executable("python");
+#endif
+
+  QTextStream out(veles::ui::LogWidget::output());
+  out
+      << "Trying to start a new server..." << endl
+      << "    working directory: " << directory_path << endl
+      << "    python script name: " << server_file_name << endl
+      << "    python interpreter executable: "
+      << python_interpreter_executable << endl
+      << "    arguments:" << endl;
+  for (const auto& arg : arguments) {
+    out << "        " << arg << endl;
+  }
+  out << endl;
+
+  server_process_->start(python_interpreter_executable, arguments);
+}
+
+void ConnectionManager::killLocalServer() {
+  if(server_process_) {
+    server_process_->terminate();
+  }
+}
+
+void ConnectionManager::disconnect() {
+  // TODO disconnect from the server
+  // Can be done once C++ client library is created.
+}
+
+void ConnectionManager::serverProcessReadyRead() {
+  char buf[10240];
+  if(server_process_ && (server_process_->canReadLine()
+      || server_process_->bytesAvailable() > (qint64)sizeof(buf))) {
+    qint64 len = server_process_->readLine(buf, sizeof(buf));
+    if(len > 0) {
+      LogWidget::output()->write(buf, len);
```

Comment:
when output contains non ASCII characters (for exmaple **Ś**witezianka) then it doesn't display corretly on windows. Also there is some buffering issues. For example output from script "Ready" is not displayed until there is more characters in output buffer